futures-core/src/lib.rs

```diff
@@ -28,4 +28,6 @@ pub mod future;
 
 pub mod stream;
 
+pub mod stream_obj;
```

Comment:
`StreamObj` needs to be available under `futures_core::stream::StreamObj` for consistency with the `future` module

To do this:
- Rename `stream.rs` to `stream/mod.rs`
- Rename `stream_obj.rs` to `stream/stream_obj.rs`
- Export `StreamObj` from `stream/mod.rs`